File: Scala/src/main/scala/PrimaryClasses/FileControl.scala
package PrimaryClasses

import java.io.{FileOutputStream, IOException}

import IExecutable.IExecutable

class FileControl extends IExecutable {
  override def Question1(): Unit = {
    try {
      val fs = new FileOutputStream("output.txt")
      val outputString = "This is the string to write.\n"
      fs.write(outputString.getBytes)
      fs.close()
      println("Completed.")
    } catch {
      case ex: IOException => println(ex.getMessage)
    }
  }

  override def Question2(): Unit = {

  }

  override def Question3(): Unit = {

  }

  override def Question4(): Unit = {

  }
}

File: Java/src/com/ecc/gkgo1a/FileControl.java
package com.ecc.gkgo1a;

import java.io.*;

public class FileControl extends IExecutable {

    @Override
    public void Question1() {
        try {
            FileOutputStream fs = new FileOutputStream("output.txt");
            String outputString = "This is the string to output.";
            fs.write(outputString.getBytes());
            fs.close();
            System.out.println("Completed.");
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }
    }

    @Override
    public void Question2() {

    }

    @Override
    public void Question3() {

    }

    @Override
    public void Question4() {

    }
}
